fondant/pipeline.py

```diff
@@ -0,0 +1,447 @@
+                component_task = _set_task_configuration(
+                    component_task, fondant_component_op
+                )
+                # Set the execution order of the component task to be after the previous
+                # component task.
+                if previous_component_task is not None:
+                    component_task.after(previous_component_task)
+
+                # Update the manifest path to be the output path of the current component task.
+                manifest_path = component_task.outputs["output_manifest_path"]
+
+                previous_component_task = component_task
+
+            return pipeline
+
+        logger.info(f"Compiling pipeline: {self.name}")
+
+        kfp.compiler.Compiler().compile(pipeline, self.package_path)
+
+        logger.info("Pipeline compiled successfully")
+
+    def __repr__(self) -> str:
+        """Return a string representation of the FondantPipeline object."""
+        return f"{self.__class__.__name__}({self._graph!r}"
+
+
+class FondantClient:
+    """Class representing a Fondant Client."""
+
+    def __init__(self, host: str):
+        """
+        Args:
+            host: The `host` URL argument specifies the Kubeflow Pipelines API endpoint to
+             which the client should send requests.
+        """
+        self.host = host
+        self.client = kfp.Client(host=self.host)
+
+    def get_pipeline_id(self, pipeline_name: str) -> str:
+        """
+        Function that returns the id of a pipeline given a pipeline name
+        Args:
+            pipeline_name: the name of the pipeline
+        Returns:
+            The pipeline id.
+        """
+        return self.client.get_pipeline_id(pipeline_name)
+
+    def get_pipeline_version_ids(self, pipeline_id: str) -> t.List[str]:
+        """Function that returns the versions of a pipeline given a pipeline id."""
+        pipeline_versions = self.client.list_pipeline_versions(pipeline_id).versions
+        return [getattr(version, "id") for version in pipeline_versions]
+
+    def delete_pipeline(self, pipeline_name: str):
+        """
+        Function that deletes the pipeline name
+        Args:
+            pipeline_name: the name of the pipeline to delete.
+        """
+        pipeline_id = self.get_pipeline_id(pipeline_name)
+        if pipeline_id is not None:
+            pipeline_version_ids = self.get_pipeline_version_ids(pipeline_id)
+            # All versions need to be first deleted
+            for pipeline_version_id in pipeline_version_ids:
+                self.client.delete_pipeline_version(pipeline_version_id)
+            self.client.delete_pipeline(pipeline_id)
+
+            logger.info(
+                f"Pipeline {pipeline_name} already exists. Deleting old pipeline..."
+            )
+        else:
+            logger.info(f"No existing pipeline under `{pipeline_name}` name was found.")
+
+    def compile_and_upload(
+        self,
+        *,
```

Comment:
Nit, but I think it makes more sense to have `pipeline` as a positional argument.